### PR TITLE
Pi: correct parsing for max length

### PR DESCRIPTION
**Pull request by kirkins**
## Description of new Instant Answer, or changes

This fixes the issue where the Pi Goodie would trigger beyond the max digits provided by the Goodie. 

The issue was related to the max digits being calculated by the length of pi, problem being that `3.` was included in the calculation of length.

## Related Issues and Discussions

#4440

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/testing-reference/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/Pi

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/Pi.pm [hunk 1]
```diff
@@ -22,12 +22,12 @@ my $PI = '3.14159265358979323846264338327950288419716939937510582097494459230781
          '147303598253490428755468731159562863882353787593751957781857780532171226806613001927876611195909216'.
          '4201989';
 
-my $PI_max_digits = length($PI);
+my $PI_max_digits = length($PI)-2;
```
Comment (ManrajGrover) [inline lib/DDG/Goodie/Pi.pm]: Nit: `length($PI) - 2` with spaces.

File: lib/DDG/Goodie/Pi.pm [hunk 2]
```diff
@@ -22,12 +22,12 @@ my $PI = '3.14159265358979323846264338327950288419716939937510582097494459230781
...
 
 handle query_raw => sub {
 
     return unless $_ =~ /^(?:pi|π)?\s*(?:to|first)?\s*(?<decimal>\d+)\s*(?:(?:decimal|digit)s?)?\s*(?:of\s+(?:pi|π))?$/i && 
-                        $+{decimal} > 0 && $+{decimal} < $PI_max_digits;
+        $+{decimal} > 0 && $+{decimal} < $PI_max_digits;
```
Comment (ManrajGrover) [inline lib/DDG/Goodie/Pi.pm]: Fix: `$+{decimal} <= $PI_max_digits` otherwise it won't trigger for `pi 1000`.

**Comment by mintsoft** [comment]
LGTM Thanks @kirkins

**Pull request merged**